common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -89,7 +92,14 @@ private static void loadConfigOptions(List<ConfigHolder> holders) {
 
             // read value and update option
             JsonElement holderJson = configObject.get(holder.getJsonName());
-            Object value = gson.fromJson(holderJson, holder.getType());
+            Object value;
+
+            if (holder.getTypeOverride() == null) {
```

Comment:
Why is this logic not part of holder.getType? Seems to me that the "override" should be internal to the ConfigHolder.

Reply:
holder.getType returns a `Class<?>` and typeOverride is an actual Type. Can I just cast the Type to Class<? extends Type>? Not sure.

Reply:
So the two calls to `fromJson` are actually overloaded with different argument types. I see. It might be that there is nothing to be done then.

I'm not sure you can cast like that, but I'm guessing you'd be able to create a Type given a `Class<?>`, so it's possible you can have getType return a Type in all situations, the overrideType if it's specified, and a Type created from the Class otherwise.